### DISCUSSION TITLE
there are unknown plugin options for "gatsby-source-shopify": shopName, accessToken

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#32891] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hello, I try to use Gatsby with Shopify, but when I pass my `shopName` and my `accesToken` I have an error message 
```
Invalid plugin options for "gatsby-source-shopify":

- "password" is required
- "storeUrl" is required
```
It's my first time with an API Shopify and with implement it in `Gatsby` 

My config is something like that 
```
module.exports = {
  siteMetadata: {
    siteUrl: `https://www.yourdomain.tld`,
  },
  plugins: [
    {
      resolve: `gatsby-source-shopify`,
      options: {
        // storeUrl: `cafe-366.myshopify.com`,
        // password: `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`,
        shopName: `cafe-366.myshopify.com`,
        accessToken: `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`,
      },
    },
  ],
};
```

I try to replace the `shopName` and the `accesToken` by the `StoreUrl `and the `accesToken` by the `password`, but that's don't work too.

For information in the Shopify Partners I create an `Custom App` and I give the authorization to activate the `API Storefront`   for the URL application I write `https://cafe-366.myshopify.com/` and for the authorized redirection I write `http://localhost:8000/` because for now I work locally. 

I cannot figure me why I cannot pass the `ShopName` and the `accesToken` like in your example.

### Reproduction Link

none

### Steps to Reproduce

none

### Expected Result

none

### Actual Result

` ERROR #11331  PLUGIN`

### Environment

```shell
OSX mojave
```

### Config Flags

none

**Comment by kamerondotcom** [comment]
I'm having this issue too. Strange to see a password required as an option :/

**Comment by LekoArts** [comment, marked as answer]
Hi!

You might have followed an outdated guide on `gatsby-source-shopify` as the error you're seeing is correct.
Here is the written form of getting started: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-shopify#getting-started

I also made a video with a colleague where I show how to set up the Shopify App and which information to use (timestamped): https://youtu.be/58OD42jyF4o?t=445

So the password I put into Gatsby Cloud there goes into `password` option of the source plugin. The password itself refers to the Admin API as the source plugin uses this under the hood.

**Reply by knupel** [reply]
Hello @LekoArts, thanks for the new track to follow. The first path I follow is this one https://www.gatsbyjs.com/docs/building-an-ecommerce-site-with-shopify/ if it's a wrong one, maybe it's necessary to remove it ?  
So now I go to investigate on your suggestion and look the video with attention. Thanks a lot.

**Reply by LekoArts** [reply]
Yeah, sorry for the outdated guide. It'll get updated with https://github.com/gatsbyjs/gatsby/pull/32884 (and possible follow-ups)

**Reply by knupel** [reply]
Hello @LekoArts, step by step I continue to dig and follow your `youtube` tutorial, it's exactly what I need, but there is another problem. I create the private app in Shopify to create the `password`, the `apiKey`, but when I pass those data in ".env" on my local project I receive the same error 
```
 ERROR #11331  PLUGIN

Invalid plugin options for "gatsby-source-shopify":

- "password" is required
- "storeUrl" is required
```


gatsby config
```
plugins: [
    {
      resolve: `gatsby-source-shopify`,
      options: {
        apiKey: process.env.SHOPIFY_API_KEY,
        password: process.env.SHOPIFY_SHOP_PASSWORD,
        storeUrl: process.env.GATSBY_SHOPIFY_STORE_URL,
        shopifyConnections: ["collections"],
      },
    },
  ],
```
I don't understand how write exactly the data in the `.env`. If I must use quote or not. If I must write the `shppa_`before the password or not... but in all case that's don't work and return this error above.

On the other hand, I manage with success `Shopify app` to my site on `Gatsby Cloud` :)

Below, a screenshot of my step. For information in French, `mot de passe` mean `password` and `clé API` mean `API key`
![env_gatsby_data](https://user-images.githubusercontent.com/3332269/130764643-ca672c0c-98d6-478f-aeb8-02e7d23ba58e.jpg)
![shopify_data](https://user-images.githubusercontent.com/3332269/130764648-28a32d69-9dd3-4f35-a3da-6adfb8a38685.jpg)

**Reply by LekoArts** [reply]
Copy the keys as it, so including `shppa_`.

Make sure that you have this at the top of your `gatsby-config.js` file:

```js
require("dotenv").config()
```

**Reply by knupel** [reply]
> ```js
> require("dotenv").config()
> ```

Et voilà... I don't know I must write this simple and incredible sentence... `REQUIRE` thanks, now that's work like a charm !  Go to the next step and problem :)

**Reply by JHoangUC** [reply]
this worked for me too thank you

**Comment by devzakir** [comment]
I have added `Admin API access token` as **`password`** and **`storeUrl`**. it works fine! 

![image](https://user-images.githubusercontent.com/33604077/155933752-675cbdb6-04f7-48f4-a7f7-605edaabdb98.png)

**Reply by abouthalf** [reply]
Thanks for sharing this - the Shopify UX has changed and the Gatsby plugin information is out of date.
